src/main/java/org/broadinstitute/hellbender/utils/haplotype/Event.java

```diff
@@ -27,14 +26,38 @@ public class Event implements Locatable {
     private static final long serialVersionUID = 1L;
 
     public Event(final String contig, final int start, final Allele ref, final Allele alt) {
-        // TODO: Should this instead silently trim to the minimal representation?
-        Utils.validateArg(ref.length() == 1 || alt.length() == 1 || differentLastBase(ref, alt), "Ref and alt alleles have same last base, hence this event is not in its minimal representation.");
         Utils.validateArg(ref.isReference(), "ref is not ref");
         this.contig = contig;
         this.start = start;
-        stop = start + ref.length() - 1;
-        refAllele = ref;
-        altAllele = alt;
+        //check for minimal representation
+        if(ref.length() == 1 || alt.length() == 1 || differentLastBase(ref.getBases(), alt.getBases())){
+            refAllele = ref;
+            altAllele = alt;
+        } else {
+            Pair<Allele, Allele> minimalAlleles = makeMinimalRepresentation(ref, alt);
+            refAllele = minimalAlleles.getLeft();
+            altAllele = minimalAlleles.getRight();
+        }
+        stop = start + refAllele.length() - 1;
+    }
+
+    /**
+     * Returns a pair of alleles that are in minimal representation (removes identical suffixes)
+     * @param ref allele not in minimal representation
+     * @param alt allele not in minimal representation
+     * @return pair of alleles in minimal representation
+     */
+    private static Pair<Allele, Allele> makeMinimalRepresentation(Allele ref, Allele alt) {
```

Comment:
Because it's much more common that the allele representation is already minimal, it's worth checking for this and just returning the original `ref` and `alt` if possible without needing to call `Allele.create`.